services/minespace-web/src/components/dashboard/mine/MineDashboard.js

```diff
@@ -59,17 +67,25 @@ export class MineDashboard extends Component {
   componentWillReceiveProps(nextProps) {
     const { activeTab, id } = nextProps.match.params;
     if (activeTab !== this.state.activeTab) {
-      this.setState({ activeTab });
+      this.setState({ activeTab: this.extractActiveTab(activeTab) });
     }
     if (!nextProps.staticContentLoadingIsComplete) {
       this.loadStaticContent();
     }
 
-    if (nextProps.match.params.id !== this.props.match.params.id) {
-      this.loadMine(id, activeTab);
+    if (
+      nextProps.match.params.id !== this.props.match.params.id ||
+      (this.extractActiveTab(activeTab) === "nods" && !this.state.isLoaded)
+    ) {
+      this.loadMine(id, this.extractActiveTab(activeTab));
     }
   }
 
+  extractActiveTab = (str) => {
```

Comment:
Can we use a more meaningful variable name than str?

Is the problem that is causing the bug essentially that instead of passing the activeTab like `?activeTab={tab}` that we're just passing it like `?{tab}`? (Or otherwise not in the format that the component expects) If so then we might want to fix that problem at the source to make it consistent, that is handled in the `NotificationDrawer` component.